application/src/main/java/org/togetherjava/tjbot/commands/tag/TagCommand.java

```diff
@@ -0,0 +1,88 @@
+package org.togetherjava.tjbot.commands.tag;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Emoji;
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent;
+import net.dv8tion.jda.api.events.interaction.SelectionMenuEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.interactions.components.Button;
+import net.dv8tion.jda.api.interactions.components.ButtonStyle;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+import net.dv8tion.jda.api.interactions.components.selections.SelectionMenu;
+import org.togetherjava.tjbot.commands.AbstractCommand;
+import org.togetherjava.tjbot.tag.TagSystem;
+
+import java.util.List;
+
+public class TagCommand extends AbstractCommand {
+    private final TagSystem tagSystem;
+
+    public TagCommand(TagSystem tagSystem) {
+        super("tag", "View a tag", true);
+
+        this.tagSystem = tagSystem;
+    }
+
+    @Override
+    public CommandData addOptions(CommandData data) {
+        return data.addOption(OptionType.STRING, "id", "Tag id")
+            .addOption(OptionType.BOOLEAN, "raw", "Raw");
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        OptionMapping idOption = event.getOption("id");
+
+        if (idOption != null) {
+            String tagId = idOption.getAsString();
+
+            TagUtility.replyTag(event, tagId, event.getUser().getAsTag(), tagSystem,
+                    event.getOption("raw") != null, event.getUser().getId());
+        } else {
```

Comment:
Prefer early-out. So `if (...) { ... return; } ...` and then the rest without a big `else` block.

Reply:
done